frontend/app_flowy/lib/workspace/presentation/home/menu/menu.dart

```diff
@@ -40,13 +40,22 @@ class HomeMenu extends StatelessWidget {
   })  : _collapsedNotifier = collapsedNotifier,
         super(key: key);
 
+  @override
+  State<HomeMenu> createState() => _HomeMenuState();
+}
+
+class _HomeMenuState extends State<HomeMenu> {
+  /// Maps the hashmap of the menu items to their index in reorderable list view.
+  //TODO @gaganyadav80: Retain this map to persist on app restarts.
+  final Map<int, int> _menuItemIndex = <int, int>{};
```

Comment:
The order should be saved using the FlowySDK. I could implement that for you later. So there is no need to cover to StatefulWidget.